### PR TITLE
[risk=low][RW-5572] Convert a few to dot-notation

### PR DESCRIPTION
**Pull request by jmthibault79**
Following up on #6281 

Remove a few dot-notation excpetions

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI and/or API server with `yarn test-local` or [yarn test-local-devup](https://github.com/all-of-us/workbench/blob/main/e2e/README.md#examples)
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: ui/src/app/pages/login/account-creation/account-creation-institution.tsx [hunk 1]
```diff
@@ -301,34 +301,26 @@ export class AccountCreationInstitution extends React.Component<Props, State> {
           message: '^Institutional role cannot be blank',
         },
       },
+      checkEmailResponse:
+        !isBlank(
```
Comment (jmthibault79) [inline ui/src/app/pages/login/account-creation/account-creation-institution.tsx:305]: same logic as below: add this check under certain conditions

File: ui/src/app/pages/workspace/workspace-edit.tsx [hunk 2]
```diff
@@ -244,15 +244,10 @@ function getDiseaseNames(keyword) {
       return response.json();
     })
     .then((matches) => {
-      const labeledMatches = fp.filter((elt) => elt.hasOwnProperty('label'))(
-        matches
+      const labeledMatches = matches.filter((elt) =>
```
Comment (jmthibault79) [inline ui/src/app/pages/workspace/workspace-edit.tsx:247]: `fp` can be replaced by non-fp `filter()` and `map()` which the linter understands better

File: ui/src/app/pages/workspace/workspace-edit.tsx [hunk 3]
```diff
@@ -243,17 +243,11 @@ function getDiseaseNames(keyword) {
...
       return response.json();
     })
-    .then((matches) => {
-      const labeledMatches = fp.filter((elt) => elt.hasOwnProperty('label'))(
-        matches
-      );
-      // Property 'label' does not exist on type 'unknown'
-      // TODO RW-5572 confirm proper behavior and fix
-      // eslint-disable-next-line @typescript-eslint/dot-notation
-      const diseases = fp.map((elt) => elt['label'])(labeledMatches);
```
Comment (jmthibault79) [inline ui/src/app/pages/workspace/workspace-edit.tsx:253]: looks like the linter didn't understand the `fp` version, but the non-fp version works.

File: ui/src/app/pages/workspace/workspace-share.spec.tsx [hunk 4]
```diff
@@ -191,15 +191,12 @@ describe('WorkspaceShare', () => {
...
-        // Property 'isDisabled' does not exist on type 'HTMLAttributes'. Did you mean 'disabled'?
-        // TODO RW-5572 confirm proper behavior and fix
-        // eslint-disable-next-line @typescript-eslint/dot-notation
-        .props()['isDisabled']
-    ).toBe(true);
+    const roleSelectProps = wrapper
+      .find('[data-test-id="harry.potter@hogwarts.edu-user-role"]')
+      .first()
+      .props() as object;
+    expect(roleSelectProps.hasOwnProperty('isDisabled')).toBe(true);
```
Comment (calbach) [inline ui/src/app/pages/workspace/workspace-share.spec.tsx:198]: nit: I don't think it's necessary to make this distinction. L199 will already handle this case and provide an equally informative error message

File: ui/src/app/pages/login/account-creation/account-creation.tsx [hunk 5]
```diff
@@ -211,6 +211,14 @@ export class AccountCreation extends React.Component<
   validate(): { [key: string]: string } {
     const { gsuiteDomain } = serverConfigStore.get().config;
 
+    // The validation data for this form is *almost* the raw Profile, except for the additional
+    // 'usernameWithEmail' field we're adding, to be able to separate our validation on the
+    // username itself from validation of the full email address. For this reason, we need to cast
+    // the profile object to 'any'.
+    const validationData = { ...this.state.profile } as any;
+    validationData.usernameWithEmail =
+      validationData.username + '@' + gsuiteDomain;
```
Comment (calbach) [inline ui/src/app/pages/login/account-creation/account-creation.tsx:220]: Existing issue, but I think we can avoid the ugliness of the `as any` by just doing the following:

```suggestion
    // username itself from validation of the full email address.
    const validationData = {
      ...this.state.profile,
      usernameWithEmail: this.state.profile.username + '@' + gsuiteDomain
    };
```

**Review by calbach — APPROVED** [review]

File: ui/src/app/pages/login/account-creation/account-creation.tsx (hunk 5 above)
Comment (jmthibault79) [reply ui/src/app/pages/login/account-creation/account-creation.tsx:220]: nice

File: ui/src/app/pages/workspace/workspace-share.spec.tsx (hunk 4 above)
Comment (jmthibault79) [reply ui/src/app/pages/workspace/workspace-share.spec.tsx]: thanks.

**Pull request merged**